dlc-manager/src/sub_channel_manager.rs

```diff
@@ -2741,8 +2824,64 @@ where
                         .map_err(|e| Error::InvalidState(format!("{:?}", e)))?;
                 }
                 SubChannelState::Confirmed(a) => {
+                    self.ln_channel_manager
+                        .with_useable_channel_lock(&channel.channel_id, peer_id, |channel_lock| {
+                            let dlc_channel_id =
+                                channel.get_dlc_channel_id(0).ok_or_else(|| {
```

Comment:
Why does using a 0 index always work?

This reminds me that we've been wanting to link the sub-channel ID to the current contract ID (perhaps also older ones too) for a while.

Reply:
The point for this parameter is to be able to extend to have multiple dlc channels in the future, but since for now there is only support for one, it's always the one with zero index. Maybe it was not the best to anticipate like this, but since it's already there I feel better to keep it.

Reply:
> This reminds me that we've been wanting to link the sub-channel ID to the current contract ID (perhaps also older ones too) for a while.

At the moment you need to first get the dlc channel id before getting the contract id, but you could probably add a utility method that just does both.